sanicap/sanicap.py

```diff
@@ -51,7 +53,7 @@ def _random_mac(self, address):
         def pad_bin(unpadded):
             return format(int('0x' + unpadded.replace(':','').replace('.',''), 16), '048b')
 
-        unmasked = ''.join([str(randint(0,1)) for x in xrange(0, 48 - self.mask)])
+        unmasked = ''.join([str(randint(0,1)) for x in range(0, 48 - self.mask)])
```

Comment:
I realize this was existing: join works with an iterator, no need to create an intermediate list
```suggestion
        unmasked = ''.join(str(randint(0,1)) for x in range(0, 48 - self.mask))
```